src/datachain/lib/dc.py

```diff
@@ -1836,6 +1841,7 @@ def parse_tabular(
         if isinstance(output, dict):
             model_name = model_name or object_name or ""
             model = dict_to_data_model(model_name, output)
+            output = model
```

Comment:
added to hit the condition below:

 `elif isinstance(output, type(BaseModel)):` 

https://github.com/iterative/datachain/pull/607/files#diff-0b1cfd3d89c057cc57811e07eb20298eeaaee60c54cc66e5128f618d144f42eeR1850

and always create here output with normalized names (`dict_to_data_model` normalizes names)